### DISCUSSION TITLE
Inconsistency in Substate ID's

### DISCUSSION THREAD

**Discussion by ari-sc** [andrew-gresyk/HFSM2#98] [Q&A]
It seems as though the substate ID's are not consistent.
The substate ID when using stateId<SUB_STATE>() is not equal to activeSubState(PARENT_STATE) when SUB_STATE is PARENT_STATE's active substate.
Is this intentional behavior?

**Comment by andrew-gresyk** [comment]
`activeSubState<TRegion>()` returns a 0-based index of an active sub-state, not global `StateID` of that state.

**Comment by andrew-gresyk** [comment]
`StateID` for the currently active sub-state can be calculated by adding `stateId<TRegion>()` and `activeSubState<TRegion>()`

**Comment by ari-sc** [comment]
It actually seems as though activeSubState<TRegion>() is 0-based but stateId<TRegion>() is 1-based. is that correct? so I actually need to add stateId + activeSubState + 1 to find the current active sub-state global ID?

**Comment by andrew-gresyk** [comment]
Both a 0-based.
Is there an example showing the behavior you describe?

**Comment by ari-sc** [comment]
Sorry I haven't responded yet. I will try to provide an example when I get a chance.
